lamindb_setup/core/_settings_storage.py

```diff
@@ -152,35 +152,39 @@ def init_storage(
         "hub-record-not-created", "hub-record-retireved", "hub-record-created"
     ] = "hub-record-not-created"
     # the below might update the uid with one that's already taken on the hub
-    if not prevent_register_hub:
-        if ssettings.type_is_cloud or register_hub:
-            from ._hub_core import delete_storage_record, init_storage_hub
-
-            hub_record_status = init_storage_hub(
-                ssettings,
-                auto_populate_instance=not init_instance,
-                created_by=created_by,
-                access_token=access_token,
-            )
-    # below comes last only if everything else was successful
-    try:
-        # (federated) credentials for AWS access are provisioned under-the-hood
-        # discussion: https://laminlabs.slack.com/archives/C04FPE8V01W/p1719260587167489
-        # if access_token was passed in ssettings, it is used here
-        mark_storage_root(ssettings.root, ssettings.uid)  # type: ignore
```

Comment:
The fact that this was always called seems like a pretty brutal logical error. I wonder how this made it into the production code. Probably very old.

Reply:
Yes, this is very old code. It looks pretty bad, i agree, but i don't see why it is brutal as it doesn't break anything actually.